plugins/woocommerce/includes/class-wc-discounts.php

```diff
@@ -627,7 +627,9 @@ protected function validate_coupon_usage_limit( $coupon ) {
 					'return'      => 'ids',
 				)
 			);
-			if ( count( $recent_pending_orders ) > 0 ) {
+
+			$recent_pending_orders_count = is_countable( $recent_pending_orders ) ? count( $recent_pending_orders ) : 0;
+			if ( $recent_pending_orders_count > 0 ) {
```

Comment:
For these where we query through `wc_get_orders()`, we can assume we will always have an array returned since we are certain the `paginate` parameter is not being set.

Reply:
Done 318fe824600808a8a6bf6b97a406d86d4bb65d7c